DSharpPlus.Commands/Processors/TextCommands/Parsing/DefaultTextArgumentSplicer.cs

```diff
@@ -0,0 +1,137 @@
+        for (; i < text.Length; i++)
+        {
+            if (!char.IsWhiteSpace(text[i]))
+            {
+                break;
+            }
+        }
+
+        startAt = i;
+
+        int endPosition = -1;
+        int startPosition = startAt;
+        for (i = startPosition; i < text.Length; i++)
+        {
+            if (char.IsWhiteSpace(text[i]) && !inQuote && !inTripleBacktick && !inBacktick && !inEscape)
+            {
+                endPosition = i;
+            }
+
+            if (text[i] == '\\' && text.Length > i + 1)
+            {
+                if (!inEscape && !inBacktick && !inTripleBacktick)
+                {
+                    inEscape = true;
+                    if (text.IndexOf("\\`", i) == i || quoteCharacters.Any(c => text.IndexOf($"\\{c}", i) == i) || text.IndexOf("\\\\", i) == i || (text.Length >= i && char.IsWhiteSpace(text[i + 1])))
+                    {
+                        removeIndices.Add(i - startPosition);
+                    }
+
+                    i++;
+                }
+                else if ((inBacktick || inTripleBacktick) && text.IndexOf("\\`", i) == i)
+                {
+                    inEscape = true;
+                    removeIndices.Add(i - startPosition);
+                    i++;
+                }
+            }
+
+            if (text[i] == '`' && !inEscape)
+            {
+                bool tripleBacktick = text.IndexOf("```", i) == i;
+                if (inTripleBacktick && tripleBacktick)
+                {
+                    inTripleBacktick = false;
+                    i += 2;
+                }
+                else if (!inBacktick && tripleBacktick)
+                {
+                    inTripleBacktick = true;
+                    i += 2;
+                }
+
+                if (inBacktick && !tripleBacktick)
+                {
+                    inBacktick = false;
+                }
+                else if (!inTripleBacktick && tripleBacktick)
+                {
+                    inBacktick = true;
+                }
+            }
+
+            if (quoteCharacters.Contains(text[i]) && !inEscape && !inBacktick && !inTripleBacktick)
+            {
+                removeIndices.Add(i - startPosition);
+
+                inQuote = !inQuote;
+            }
+
+            if (inEscape)
+            {
+                inEscape = false;
+            }
+
+            if (endPosition != -1)
+            {
+                startAt = endPosition;
+                return startPosition != endPosition ? CleanupString(text[startPosition..endPosition], removeIndices) : null;
+            }
+        }
+
+        startAt = text.Length;
+        return startAt != startPosition ? CleanupString(text[startPosition..], removeIndices) : null;
+    }
+
+    private static string CleanupString(string s, IList<int> indices)
+    {
+        if (!indices.Any())
+        {
+            return s;
+        }
+
+        int li = indices.Last();
+        int ll = 1;
```

Comment:
I think this time its my turn to yell at you for var naming

Reply:
Nope, you don't. As I've repeated several times already, this is copy and pasted from CNext.

Reply:
I would like it if you change them. We all want to improve the code quality

Reply:
The text argument slicer needs to be refactored anyways because of previous comments. As I've previously stated, I copied CommandsNext text argument slicer into DSP.Commands because I needed to test other features. I have been requesting for help with the refactor multiple times.

Reply:
Hey, please don't feel offended. Overall you did very good with this extension and most of my comments are little things. I havent read all comments on the pr because i have not much time atm.

Reply:
I'm sorry, I was out of line. Thank you for your constructive comments, I'll be sure to address them all in a better tone.